### PR TITLE
Do not use instanceid for reservation id

### PR DESCRIPTION
**Pull request by alenkacz**
<<Replace this line with your Revision Title>>

Summary:
For more details follow http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html.

JIRA issues:

### CODE REVIEW

**Comment by meln1k** [comment]
This change looks scary. I'd like to see some integration tests where we test an upgrade process.

File: src/main/scala/mesosphere/marathon/core/instance/Reservation.scala [hunk 1]
```diff
@@ -63,8 +63,7 @@ object Reservation {
     def apply(label: String): Id = label match {
       case SimplifiedIdRegex(safeRunSpecId, prefix, uuid) =>
         val runSpec = PathId.fromSafePath(safeRunSpecId)
-        val instanceId = Instance.Id(runSpec, Prefix.fromString(prefix), UUID.fromString(uuid))
-        SimplifiedId(instanceId)
+        LegacyId(runSpec, "", UUID.fromString(uuid))
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/core/instance/Reservation.scala]: I'm not sure this is right. This is basically the factory method. When we match the `SimplifiedId` we should not return the `LegacyId`. What do you do if the reservation id is a `SimplifiedId`?

Comment (alenkacz) [reply src/main/scala/mesosphere/marathon/core/instance/Reservation.scala]: yeah I don't think this is right either... That was pre-lunch thing and I touched different part of code than I wanted to :) now it should be good... that's the part of code that `InstanceOpFactoryImpl` uses

**Comment by alenkacz** [comment]
oh yeah, but this won't work if someone already migrated to 1.8 and has marathon with new reservation created via the `SimplifiedId` :/

**Pull request closed without merging**